src/Microsoft.AspNetCore.Server.IIS/Core/IISEventSource.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.Server.IIS.Core
+{
+    [EventSource(Name = "Microsoft-AspNetCore-Server-IIS")]
+    internal sealed class IISEventSource : EventSource
+    {
+        public static readonly IISEventSource Log = new IISEventSource();
+
+        private IISEventSource()
+        {
+        }
+
+        [NonEvent]
+        public void RequestStart(IISHttpContext httpContext)
+        {
+            // avoid allocating the trace identifier unless logging is enabled
+            if (IsEnabled())
+            {
+                RequestStart(httpContext.ConnectionId.ToString(), httpContext.TraceIdentifier);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        [Event(3, Level = EventLevel.Verbose)]
+        private void RequestStart(string connectionId, string requestId)
+        {
+            WriteEvent(3, connectionId, requestId);
+        }
+
+        [NonEvent]
+        public void RequestStop(IISHttpContext httpContext)
```

Comment:
You could consider updating some EventCounters here as well so that users can get the low-granularity data too. See [SignalR's counters](https://github.com/aspnet/SignalR/blob/release/2.2/src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionsEventSource.cs) for an example that's low-allocating and relatively performance-friendly.